src/factories/node.ts

```diff
@@ -180,6 +180,12 @@ export async function nodeContainerFactory(node: RunGraphNode) {
   function getNodeCacheKey(nodeData: RunGraphNode): string {
     const endTime = nodeData.end_time ?? new Date()
     const artifactCount = nodeData.artifacts?.length ?? 0
+    const artifactCacheKey = nodeData.artifacts?.map(artifact => {
+      if (artifact.type === 'progress') {
+        return `${artifact.id}-${artifact.data}`
+      }
+      return ''
+    }).join()
```

Comment:
We don't really need both of these. Could you write an artifacts cache key that has a value for every artifact? I think if you just return the `artifact.id` where you're doing `return ''` that'll satisfy all artifacts. 

It also be nice if we formatted this a bit for future is if we're debugging this key. Maybe join with a separator like `,` or `|`.

Reply:
If the artifact type isn't progress the `data` field won't be present. I did make a change so we don't need artifactCount, though